share/smack/top.py

```diff
@@ -899,85 +979,133 @@ def verify_bpl(args):
         return
 
     elif args.verifier == 'boogie' or args.modular:
-        command = ["boogie"]
-        command += [args.bpl_file]
-        command += ["/doModSetAnalysis"]
-        command += ["/useArrayTheory"]
-        command += ["/timeLimit:%s" % args.time_limit]
-        command += ["/errorLimit:%s" % args.max_violations]
+        command = invoke_boogie(args)
         command += ["/proverOpt:O:smt.array.extensional=false"]
         command += ["/proverOpt:O:smt.qi.eager_threshold=100"]
         command += ["/proverOpt:O:smt.arith.solver=2"]
-        if not args.modular:
-            command += ["/loopUnroll:%d" % args.unroll]
-        if args.solver == 'cvc4':
-            command += ["/proverOpt:SOLVER=cvc4"]
-        elif args.solver == 'yices2':
-            command += ["/proverOpt:SOLVER=Yices2"]
 
     elif args.verifier == 'corral':
-        command = ["corral"]
-        command += [args.bpl_file]
-        command += ["/tryCTrace", "/noTraceOnDisk", "/printDataValues:1"]
-        command += ["/k:%d" % args.context_bound]
-        command += ["/useProverEvaluate"]
-        command += ["/timeLimit:%s" % args.time_limit]
-        command += ["/cex:%s" % args.max_violations]
-        command += ["/maxStaticLoopBound:%d" % args.loop_limit]
-        command += ["/recursionBound:%d" % args.unroll]
-        command += ["/bopt:proverOpt:O:smt.qi.eager_threshold=100"]
-        command += ["/bopt:proverOpt:O:smt.arith.solver=2"]
-        if args.solver == 'cvc4':
-            command += ["/bopt:proverOpt:SOLVER=cvc4"]
-        elif args.solver == 'yices2':
-            command += ["/bopt:proverOpt:SOLVER=Yices2"]
+        command = invoke_corral(args)
 
     elif args.verifier == 'symbooglix':
-        command = ['symbooglix']
-        command += [args.bpl_file]
-        command += ["--file-logging=0"]
-        command += ["--entry-points=%s" % ",".join(args.entry_points)]
-        command += ["--timeout=%d" % args.time_limit]
-        command += ["--max-loop-depth=%d" % args.unroll]
+        command = invoke_symbooglix(args)
 
     if args.verifier_options:
         command += args.verifier_options.split()
 
     verifier_output = try_command(command, timeout=args.time_limit)
-    verifier_output = transform_out(args, verifier_output)
-    result = verification_result(verifier_output, args.verifier)
+    return process_verifier_output(args, verifier_output)
 
-    if args.json_file:
-        with open(args.json_file, 'w') as f:
-            f.write(json_output_str(result, verifier_output, args.verifier))
 
-    if result in VResult.ERROR:
-        error = error_trace(verifier_output, args.verifier)
+def get_result(result):
```

Comment:
What is the purpose of this callback function you introduced?
It seems to me that it doesn't really do anything, but I must be missing something here.